tests/Http/Offer/PriceInfoDuplicateNameValidatingRequestBodyParserTest.php

```diff
@@ -117,4 +117,65 @@ public function it_throws_on_same_names(): void
             fn () => $this->priceInfoDuplicateNameValidatingRequestBodyParser->parse($request)
         );
     }
+
+    /**
+     * @test
+     */
+    public function it_throws_on_same_names_with_different_spacing(): void
+    {
+        $priceInfo = (object) [
```

Comment:
Is it really necessary to add all the object casting?
It adds a lot of overhead syntax and is less readable

Reply:
Another option would be to create the `$priceInfo` variable as a `StdClass` object.

Sometimes you will also see this combination `json_decode(json_encode($array));`

Reply:
Casting is necessary, alternatively 1 private function can be used instead of all the different casts:
```
        $request = $this->requestBuilder
            ->build('POST')
            ->withParsedBody($this->castPriceInfoArrayToObject($priceInfo));
...
    private function castPriceInfoArrayToObject(array $priceInfoArray): object
    {
        $priceInfo = (object)$priceInfoArray;
        $priceInfo->priceInfo = array_map(fn ($item) => (object)$item, $priceInfo->priceInfo);
        return $priceInfo;
    }
```
@LucWollants, @grubolsch Do you have any preference?

Reply:
The my preference is for json_decode(json_encode($array)), I'll make the changes.